faiss/utils/hamming-inl.h

```diff
@@ -226,19 +227,52 @@ struct HammingComputerDefault {
     }
 
     void set (const uint8_t *a8, int code_size) {
-        a =  a8;
-        n = code_size;
+        this->a8 = a8;
+        quotient8 = code_size / 8;
+        remainder8 = code_size % 8;
     }
 
     int hamming (const uint8_t *b8) const {
         int accu = 0;
-        for (int i = 0; i < n; i++)
-            accu += popcount64 (a[i] ^ b8[i]);
+
+        auto a64 = reinterpret_cast<const uint64_t *>(a8);
+        auto b64 = reinterpret_cast<const uint64_t *>(b8);
+        int i = 0, len = quotient8;
+        switch (len & 7) {
+            default:
```

Comment:
I did not know that jumping there from a switch was possible...

Reply:
I refer to the [Duff's device](https://en.wikipedia.org/wiki/Duff%27s_device) here.